src/Uno.UI/UI/Xaml/Media/Imaging/SvgImageSource.cs

```diff
@@ -144,10 +144,18 @@ private async Task<ImageData> ReadFromStreamAsync(Stream stream, CancellationTok
 			stream.Position = 0;
 		}
 
-		var memoryStream = new MemoryStream();
-		await stream.CopyToAsync(memoryStream, 81920, ct);
-		var data = memoryStream.ToArray();
-		return ImageData.FromBytes(data);
+		if (stream is not MemoryStream memoryStream)
+		{
+			memoryStream = new MemoryStream();
+			await stream.CopyToAsync(memoryStream, 81920, ct);
+		}
+
+		if (!memoryStream.TryGetBuffer(out var buffer))
+		{
+			buffer = new ArraySegment<byte>(memoryStream.ToArray());
+		}
+
+		return ImageData.FromBytes(buffer);
```

Comment:
The whole idea is in this block of code. Two optimizations are employed here:

- If given MemoryStream already, no need to create a new one and do the copying.
- Instead of calling ToArray which will create a new array, just get access to the internal array. Note that this has a behavior difference in that if the original array used to create the MemoryStream was mutated, that would now be reflected in ImageData were previously it wouldn't.

@MartinZikmund thoughts?

Reply:
It may be a problem to mutate what has been provided by the user, particularly if it does not reflect what WinUI does.

Reply:
Yeah I think we need to figure out what happens exactly in WinUI